lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,618 @@
+<?php
+include(ABSPATH . "wp-admin/includes/admin.php");
+
+/**
+ * Font library class.
+ *
+ * @package gutenberg
+ * @since x.x.x
+ * 
+ */
+class WP_Fonts_Library {
+    public function __construct() {
+        $this->wp_fonts_dir = path_join( WP_CONTENT_DIR, 'fonts' );
+        $this->relative_fonts_path = content_url('/fonts/');
+
+		$this->rest_base        = 'fonts_library';
+		$this->namespace        = 'wp/v2';
+
+        $this->post_type        = 'wp_fonts_library';
+        $this->post_name        = 'wp_fonts_library';
+        $this->post_title       = 'Fonts Library'; // Do not translate
+        $this->post_status      = 'publish';
+
+        $this->base_post_query  = array  (
+            'post_type'      => $this->post_type,
+            'post_title'     => $this->post_title,
+            'post_status'    => $this->post_status,
+            'post_name'      => $this->post_name,
+            'name'           => $this->post_name,
+            'posts_per_page' => 1,
+        );
+	}
```

Comment:
This property should all be defined. PHP 8.1+ does not support dynamic properties.

Reply:
added here: https://github.com/WordPress/gutenberg/pull/50927/commits/95ced36dbc6185b31435371ff9073a5ae60f8268